packages/next/src/shared/lib/match-local-pattern.ts

```diff
@@ -0,0 +1,29 @@
+import type { LocalPattern } from './image-config'
+import { makeRe } from 'next/dist/compiled/picomatch'
+
+// Modifying this function should also modify writeImagesManifest()
+export function matchLocalPattern(pattern: LocalPattern, url: URL): boolean {
+  if (pattern.search !== undefined) {
+    if (pattern.search !== url.search) {
```

Comment:
Does ordering on the `search` value here matter since the user prop can't control intermediaries like CDNs changing it?

Reply:
Yes the order matters because `search` is a string literal.

I don't think CDN's changing it will be very common since this is going to be the value passed to the `src` prop on the image component.

In the future, we could add support for any order using an an object or array.